neurolib/models/multimodel/builder/adex.py

```diff
@@ -0,0 +1,944 @@
+                ),
+            ),
+        ]
+
+    def firing_rate_lookup(self, y, current_mu, current_sigma):
+        """
+        Translate mean and std. deviation of the current to firing rate using
+        linear-nonlinear lookup table for AdEx.
+        """
+        return _table_lookup(
+            current_mu,
+            current_sigma,
+            self.sigma_range,
+            self.mu_range,
+            self.d_sigma,
+            self.d_mu,
+            self.firing_rate_cascade,
+        )
+
+    def voltage_lookup(self, y, current_mu, current_sigma):
+        """
+        Translate mean and std. deviation of the current to voltage using
+        linear-nonlinear lookup table for AdEx.
+        """
+        return _table_lookup(
+            current_mu, current_sigma, self.sigma_range, self.mu_range, self.d_sigma, self.d_mu, self.voltage_cascade,
+        )
+
+    def tau_lookup(self, y, current_mu, current_sigma):
+        """
+        Translate mean and std. deviation of the current to tau - membrane time
+        constant using linear-nonlinear lookup table for AdEx.
+        """
+        return _table_lookup(
+            current_mu, current_sigma, self.sigma_range, self.mu_range, self.d_sigma, self.d_mu, self.tau_cascade,
+        )
+
+
+class ExcitatoryAdExMass(AdExMass):
+    """
+    Excitatory AdEx neural mass. Contains firing rate adaptation current.
+    """
+
+    name = "AdEx mean-field excitatory mass"
+    label = f"AdExMass{EXC}"
+    num_state_variables = 7
+    coupling_variables = {6: f"q_mean_{EXC}"}
+    mass_type = EXC
+    state_variable_names = [
+        "I_mu",
+        "I_A",
+        "I_syn_mu_exc",
+        "I_syn_mu_inh",
+        "I_syn_sigma_exc",
+        "I_syn_sigma_inh",
+        "q_mean",
+    ]
+    required_couplings = [
+        "node_exc_exc",
+        "node_exc_exc_sq",
+        "node_exc_inh",
+        "node_exc_inh_sq",
+        "network_exc_exc",
+        "network_exc_exc_sq",
+    ]
+    required_params = [
+        "K_exc",
+        "K_inh",
+        "c_global",
+        "K_exc_global",
+        "tau_syn_exc",
+        "tau_syn_inh",
+        "sigma_ext",
+        "J_exc_max",
+        "J_inh_max",
+        "tau_m",
+        "C_m",
+        "ext_current",
+        "ext_rate",
+        "a",
+        "b",
+        "E_A",
+        "tau_A",
+        "lambda",
+    ]
+
+    def __init__(self, params=None, lin_nonlin_cascade_filename=None, seed=None):
+        super().__init__(
+            params=params or DEFAULT_PARAMS_EXC, lin_nonlin_cascade_filename=lin_nonlin_cascade_filename, seed=seed
+        )
+
+    def _initialize_state_vector(self):
+        """
+        Initialize state vector.
+        """
+        np.random.seed(self.seed)
+        self.initial_state = (
+            np.random.uniform(0, 1, self.num_state_variables) * np.array([3.0, 200.0, 0.5, 0.5, 0.001, 0.001, 0.01])
+        ).tolist()
+
+    def _compute_couplings(self, coupling_variables):
```

Comment:
dynamics definition for AdEx is kinda ugly and long so I defined some helper convenience functions such as this - this just computes coupling - both exc -> exc and inh->exc and their squared variants